### ISSUE TITLE
Add Vimeo scraper

### ISSUE THREAD

**Issue by vheuken** [thibaudgg/video_info#95]
PR #87 updates the Vimeo API from v2 to v3, which requires an API key.

We should implement a scraper to be used when no Vimeo API key is provided, like we do with Youtube.

**Comment by vheuken** [comment]
@thibaudgg Is this going to block the next release? If nobody else jumps on it, I may or may not be able to work on it this weekend.

**Comment by vheuken** [comment]
@zdb Would you be interested in giving this a shot?

**Comment by thibaudgg** [comment]
@vheuken I'm still waiting on the jruby fixes for the next release (WDYT?), having the Vimeo scraper would be great for an X jump (2.4.2 -> 3.0.0).

**Comment by zdb** [comment]
@vheuken Sure.  I'll look into it this weekend.

**Comment by vheuken** [comment]
Thanks!

**Comment by vheuken** [comment]
Also, @thibaudgg I think the jruby fixes can wait if it works on your machine under jruby 9. They work fine on my machine, so if it is just a thing with Travis-CI for some reason I don't think it's a big deal. Plus, we have jruby under "allowed failures" and the brand new jruby implements Ruby 2.1, which we aren't officially supporting.

**Comment by thibaudgg** [comment]
@vheuken agree, don't loose time on the jruby travis failures. Once Vimeo scraper is here we can release a new version!

**Comment by vheuken** [comment]
Reminder to have a look at #101 when I tackle this.

**Comment by vheuken** [comment]
Started working on it in the branch `vimeo-scraper`

**Comment by vheuken** [comment]
Heads up: It looks like we can't get the view count without the API, so the scraper will be missing that.

**Comment by vheuken** [comment]
Alright, here's what's left:

1. `#date` and `#duration` need to be implemented
2. `#view_count` needs to be handled gracefully
3. `#available?` needs to handle nonexistant videos and password protected videos
4. VimeoPlaylist needs to be handled

**Comment by aryalsan** [comment]
Great.

**Comment by vheuken** [comment]
Alright, just gotta fix `#available`, implement VimeoPlaylist scraper, and do cleanup.

**Comment by maia** [comment]
@vheuken Any news on the status quo? The [link_thumbnailer](https://github.com/gottfrois/link_thumbnailer/issues/69) gem has an issue since September 2015 that can't be fixed without an update of video_info (there's a branch, but I'd prefer using a release version over a github branch).
 
Do you need any assistance the community could provide? Or is it an option to release the current status without additional features with a new version point number?

**Comment by vheuken** [comment]
@maia I actually started working on it again a couple weeks ago. I believe I finished up the Vimeo scraper, but I haven't started the VimeoPlaylist scraper (also there's some cleanup that still needs to be done). If you or someone you know would like to help, they can work off of the `vimeo-scraper` branch,, though at this point there isn't much left to do. I'll try my best to get to it this week.

Sorry for the delay!

**Comment by maia** [comment]
@vheuken Thanks, I'm looking forward to it. And in case it's an option to release the status quo without vimeo playlist support, I'd be fine with that too.

**Comment by maia** [comment]
@vheuken Hello again – is there any news on a new release, even without playlist support? Thanks!

**Comment by vheuken** [comment]
I'm planning on finishing the Vimeo playlist scraper tomorrow night. Sorry for the lack of updates.

**Comment by vheuken** [comment]
Hm, bit of an issue. All of the VimeoPlaylist API specs are failing. Might take me a big longer than I anticipated. Should be done this week though.

**Comment by vheuken** [comment]
@thibaudgg There is a bit of an issue. It looks like the /albums/ API endpoint has been deprecated. I could be wrong, but it may have never been included in the v3 API and it just happened to work before. It looks like all the current API provides is the ability to get a list of a user's albums. I think I can still do the scraper, but the API part will have to be dropped.

**Comment by maia** [comment]
Albums don't seem to be used much. I might have biased data, but out of all urls with vimeo.com as domain, less than 1% contain `vimeo.com/album`, and some of these are `vimeo.com/album/…/video/…`. And if vimeo dropped support, we may assume they are phasing out that feature?

**Comment by vheuken** [comment]
I honestly haven't even heard of a Vimeo album/playlist until a few months ago when I started helping out with this library, so I think you might be right. Also, the HTML changed and broke one thing for the Vimeo scraper (should be a simple fix), so I'm really wary about adding in another scraper.

It would really suck if someone depends on this feature and it was dropped, but I mean, right now it doesn't even work and I don't think anyone has noticed.

I'm going to wait of @thibaudgg on this one. I'll clean up the Vimeo scraper and fix the new issue right now. If @thibaudgg wants to drop it, then the new version should be ready to go and he'll just have to upload the gem.

**Comment by thibaudgg** [comment]
I see no issues about dropping a deprecated Vimeo feature that doesn't even work in our gem. 
Please remove it. Thanks!

**Comment by vheuken** [comment]
Great.

**Comment by vheuken** [comment]
Should be done and in master. Just waiting for Travis-CI to do it's thing before I close this.

**Comment by vheuken** [comment]
Travis passes!
